compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -2214,6 +2213,22 @@ static void setLoweringConfigForComputeOps(func::FuncOp entryPointFn,
     TileSizesListType tileSizesList = {distTileSizes, tileAndFuseSizes};
     TypeSwitch<Operation *>(op)
         .Case<tensor::PackOp>([&](auto packOp) {
+          ArrayRef<int64_t> innerTiles = packOp.getStaticInnerTiles();
+          ArrayRef<int64_t> dimPos = packOp.getInnerDimsPos();
+          auto outerDimsPerm = packOp.getOuterDimsPerm();
+          // Scale the outer dim tiles for pack op.
+          for (int i = 0, e = std::min<int>(tileSizesList.size(), 2); i < e;
```

Comment:
Could you use `TilingConfig` to get access to the tile size information? Otherwise, this will fail as we change the number of tile sizes in lowering config.

Reply:
The `tileSizesList` here is a new tile size config we want to build (in line 2214, we first set the distribution tile sizes and the vector common tile sizes). And in below it pushes the reduction tile sizes and inner vector tile sizes to make it an 4 elements tiling config.

I'm not clear how `TilingConfig` will help here since IIUC it only provides an interface to access tiling sizes instead of a builder. Would you mind giving an example?

The parts we can refactor with `TilingConfig` are line 2187, 2188 and line 2056, which currently we assume first two elements in the list are distribution and common vector tile sizes. Although they are not touched in this change, I'm open to refactor them in this change.

Reply:
SG then. I got confused by lines 2187 and 2188. I'll have to change this anyways so I can take care of this later.

Reply:
(drive by... i looked at `TilingConfig` recently and I am totally confused by it.. just putting this out there that I might have more questions on this later on).

Reply:
Happy to answer any specific question. `TilingConfig` is just an interface that provides encapsulation to all the different interpretations of lowering config that we had scattered all over KernelDispatch and other files... so the way to improve that is actually improving lowering config or the way we keep tile sizes and optimization knobs around